cpp/log/etcd_consistent_store-inl.h

```diff
@@ -56,6 +62,15 @@ EtcdConsistentStore<Logged>::~EtcdConsistentStore() {
   serving_sth_watch_task_.Cancel();
   VLOG(1) << "Waiting for watch tasks to return.";
   serving_sth_watch_task_.Wait();
+  VLOG(1) << "Joining cleanup thread";
+  {
+    std::lock_guard<std::mutex> lock(mutex_);
+    exiting_ = true;
+  }
+  serving_sth_cv_.notify_all();
+  clean_up_thread_->join();
+  clean_up_thread_.reset();
```

Comment:
Not *really* necessary, right? It's fine to leave like this, just checking that I'm not missing some subtle thing...